### ISSUE TITLE
Possible memory leak with TensorArray

### ISSUE THREAD

**Issue by awni** [tensorflow/tensorflow#1660]
I'm not sure if the usage is incorrect (see below), but I am seeing a memory leak in cpu ram when using the `TensorArray`. Seems to be consistent regardless of device placement. I noticed this because I've been using the `dynamic_rnn`.  Happy to provide further info if needed.

### Environment info
Operating System:
Ubuntu 14.04

If installed from sources, provide the commit hash:

`35c3f16f81b6a1d39a004c97160c22b73d3798e5`

### Steps to reproduce

Run this in a script  (I had to add `from tensorflow.python.ops import tensor_array_ops` to `tensorflow/python/__init__.py`)

```
import tensorflow as tf

def ta_mem_leak():

    ta = tf.python.tensor_array_ops.TensorArray(
                        dtype=tf.float32, size=10000)
    ta = ta.write(0, [0.])
    final_outputs = ta.read(0)
    return [final_outputs]

if __name__ == "__main__":
    with tf.Graph().as_default(), tf.Session() as session:
        ops = ta_mem_leak()
        for i in xrange(200000):
            res = session.run(ops)                          
```

**Comment by awni** [comment]
FYI `Unref`ing the `tensor_array` object after getting them out of the `resource_mgr` seemed to do the trick for me - https://github.com/awni/tensorflow/commit/af8bef11151ff98137c9b5ce1a27e543f311e7d9

I can PR this if it looks like the right fix.

**Comment by ebrevdo** [comment]
This does look correct.  Send a PR.  Make sure the rnn tests continue to pass.

**Comment by awni** [comment]
@ebrevdo Sounds good, I submitted the PR #1691. Please review.

**Comment by mrry** [comment]
I'm assuming this can be closed since #1691 is merged now.
